README.Rmd

```diff
@@ -34,93 +104,275 @@ pak::pkg_install("cmu-delphi/epipredict@main")
+  autoplot(
+    case_rate,
+    death_rate,
+    .color_by = "none"
+  ) +
+  facet_grid(
+    rows = vars(.response_name),
+    cols = vars(geo_value),
+    scale = "free"
+  ) +
+  geom_vline(aes(xintercept = forecast_date)) +
+  geom_text(
+    data = forecast_date_label,
+    aes(x = dates, label = "forecast\ndate", y = heights),
+    size = 3, hjust = "right"
+  ) +
+  scale_x_date(date_breaks = "3 months", date_labels = "%Y %b") +
+  theme(axis.text.x = element_text(angle = 90, hjust = 1))
+```
+</details>
+```{r show-processed-data, warning=FALSE, echo=FALSE}
+processed_data_plot
 ```
 
-To create and train a simple auto-regressive forecaster to predict the death rate two weeks into the future using past (lagged) deaths and cases, we could use the following function.
+To make a forecast, we will use a simple “canned” auto-regressive forecaster to
+predict the death rate four weeks into the future using lagged[^3] deaths and
+cases.
+
+[^3]: lagged by 3 in this context meaning using the value from 3 days ago.
 
 ```{r make-forecasts, warning=FALSE}
-two_week_ahead <- arx_forecaster(
-  covid_case_death_rates,
+four_week_ahead <- arx_forecaster(
+  covid_case_death_rates |> filter(time_value <= forecast_date),
   outcome = "death_rate",
   predictors = c("case_rate", "death_rate"),
   args_list = arx_args_list(
     lags = list(c(0, 1, 2, 3, 7, 14), c(0, 7, 14)),
-    ahead = 14
+    ahead = 4 * 7,
+    quantile_levels = c(0.1, 0.25, 0.5, 0.75, 0.9)
   )
 )
-two_week_ahead
+four_week_ahead
 ```
 
-In this case, we have used a number of different lags for the case rate, while
-only using 3 weekly lags for the death rate (as predictors). The result is both
-a fitted model object which could be used any time in the future to create
-different forecasts, as well as a set of predicted values (and prediction
-intervals) for each location 14 days after the last available time value in the
-data.
+In our model setup, we are defining as our predictors case rate lagged 0-3
+days, one week, and two weeks, and death rate lagged 0-2 weeks.
+The result `four_week_ahead` is both a fitted model object which could be used
+any time in the future to create different forecasts, and a set of predicted
+values (and prediction intervals) for each location 28 days after the forecast
+date.
 
-```{r print-model}
-two_week_ahead$epi_workflow
+Plotting the prediction intervals on the true values for our location subset[^2]:
+
+[^2]: Alternatively, you could call `autoplot(four_week_ahead)` to get the full
+    collection of forecasts. This is too busy for the space we have for plotting
+    here.
+
+<details>
+<summary> Plot </summary>
+```{r plotting_forecast, warning=FALSE}
+epiworkflow <- four_week_ahead$epi_workflow
+restricted_predictions <-
+  four_week_ahead$predictions |>
+  rename(time_value = target_date, value = .pred) |>
+  mutate(.response_name = "death_rate")
+forecast_plot <-
+  four_week_ahead |>
+  autoplot(plot_data = cases_deaths) +
```

Comment:
issue: the `plot_data` arg is not recognized. This is using `epipredict:::autoplot.canned_epipred`, right?

Reply:
It's included in this PR, my guess is your renv is using dev rather than this. either `load_all` or `renv::install(".")` should do it